### PR TITLE
Replace clap with custom CLI parser

### PR DESCRIPTION
**Pull request by marlonbaeten**
Replaced the CLI parser based on clap with a custom parser and removed clap as a dependency.
Note that the remaining integration test for the cli has been enabled and the new implementations is shorter (in loc) than the previous version using clap.

### CODE REVIEW

**Comment by marlonbaeten** [comment]
FYI I expect to do another iteration over this code. So please be critical while reviewing this.

File: lib/sudo-cli/src/lib.rs [hunk 1]
```diff
@@ -244,220 +75,268 @@ pub struct SudoOptions {
...
+            match arg.as_str() {
+                "--" => {
+                    processed.push(SudoArg::Rest(arg_iter.collect()));
+                    break;
+                }
+                long_arg if long_arg.starts_with("--") => {
+                    if long_arg.contains('=') {
+                        // convert assignment to normal tokens
+                        let (key, value) = long_arg.split_once('=').unwrap();
+                        processed.push(SudoArg::Argument(key.to_string(), value.to_string()));
```
Comment (rnijveld) [inline lib/sudo-cli/src/lib.rs:129]: Should we not check for Self::TAKES_ARGUMENT here as well? Otherwise someone might do `--background=false` and think that this will turn that option off.

File: lib/sudo-cli/src/lib.rs [hunk 2]
```diff
@@ -244,220 +75,268 @@ pub struct SudoOptions {
...
+            .peekable();
+
+        for arg in arg_iter {
+            match arg {
+                SudoArg::Flag(flag) => match flag.as_str() {
+                    "-A" | "--askpass" => {
+                        options.askpass = true;
+                    }
+                    "-b" | "--background" => {
+                        options.background = true;
```
Comment (rnijveld) [inline lib/sudo-cli/src/lib.rs:235]: I'm not sure this is the right time, but seeing we don't (yet) support these kinds of arguments, do we want to provide any user feedback at this point indicating that we don't?

File: lib/sudo-cli/src/lib.rs [hunk 3]
```diff
@@ -244,220 +75,268 @@ pub struct SudoOptions {
...
     }
 
-    pub fn parse() -> Self {
-        match Self::try_parse_from(std::env::args()) {
-            Ok(options) => options,
-            Err(err) => {
-                eprintln!("{err}");
-                exit(1);
+    /// parse an iterator over command line arguments
+    pub fn try_parse_from<I, T>(iter: I) -> Result<Self, &'static str>
```
Comment (rnijveld) [inline lib/sudo-cli/src/lib.rs:218]: Given that this function is now very generic and that we no longer have clap their test suite, I think this can do with a few unit tests showcasing proper handling of both common examples as well as some edge cases.

**Review by rnijveld — CHANGES_REQUESTED** [review]

**Review by squell — APPROVED** [review]
I see a few nitpicks that I'll edit.

Note: there are some options in there that I doubt we are going to support soon, but a.) it doesn't need to be perfect b.) I'm looking more at the overall framework of this; c.) seeing how easily --askpass was removed makes me confident we can maintain this ad-hoc CLI parser, so I'm :+1: here; and d.) the trade-offs in dependency reduction (25% less!) are worth it.
